src/contract-testing/txma-ais-consumer.pact.test.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Matchers,
+  MessageConsumerPact,
+  synchronousBodyHandler,
+  LogLevel,
+} from '@pact-foundation/pact';
+import {COMPONENT_ID} from "../data-types/constants";
+import { term } from "@pact-foundation/pact/src/dsl/matchers";
+import { boolean, number, string } from "@pact-foundation/pact/src/v3/matchers";
+import { validateEventAgainstSchema } from "../services/validate-event";
+const { like } = Matchers;
+
+const path = require('path');
+const LOG_LEVEL = process.env['LOG_LEVEL'] || 'ERROR';
+
+describe('TxMA & AIS - Contract Testing - Consumer', () => {
+  const messagePact = new MessageConsumerPact({
+    consumer: COMPONENT_ID,
+    dir: path.resolve(process.cwd(), 'pacts'),
+    pactfileWriteMode: 'update',
+    provider: 'TxMA',
+    logLevel: LOG_LEVEL as LogLevel,
+  });
+
+  describe('Incoming event is received from TxMA', () => {
+    it('accepts a valid intervention event from TxMA', () => {
+      return (
+        messagePact
+          .given('AIS is healthy')
+          .expectsToReceive('a valid intervention event')
+          .withContent({
+            timestamp: like(1705318190),
+            event_name: 'TICF_ACCOUNT_INTERVENTION',
+            user: {
+              user_id: string('urn:fdc:gov.uk:2022:USER_ONE')
+            },
+            extensions: {
+              intervention : {
+                intervention_code: term({ generate: '01', matcher: '^(01|02|03|04|05|06|07)$'}),
```

Comment:
Apols for the stupid questions, its basically just to deepen my understanding, had to go and read the pact docs there due to never seeing pact testing before, but my questions are, are we just essentially testing to see if the services talk to each other, hence why we are only generating `01` and then matching it against the regex of the matcher, then verifying the result? As we are just testing the connection (i think) is that why we dont need to generate multiple different codes here? And what does `term` do, is it just a constructor to enable the generation and matcher?